bundles/org.openhab.binding.anel/src/main/java/org/openhab/binding/anel/internal/state/AnelStateUpdater.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.anel.internal.state;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.anel.internal.IAnelConstants;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * Get updates for {@link AnelState}s.
+ *
+ * @author Patrick Koenemann - Initial contribution
+ */
+@NonNullByDefault
+public class AnelStateUpdater implements IAnelConstants {
+
+    public @Nullable State getChannelUpdate(String channelId, @Nullable AnelState state) {
+        if (state == null) {
+            return null;
+        }
+
+        final int index = IAnelConstants.getIndexFromChannel(channelId);
+        if (index >= 0) {
+
+            if (IAnelConstants.CHANNEL_RELAY_NAME.contains(channelId)) {
+                return getStringState(state.relayName[index]);
+            }
+            if (IAnelConstants.CHANNEL_RELAY_STATE.contains(channelId)) {
+                return getSwitchState(state.relayState[index]);
+            }
+            if (IAnelConstants.CHANNEL_RELAY_LOCKED.contains(channelId)) {
+                return getSwitchState(state.relayLocked[index]);
+            }
+
+            if (IAnelConstants.CHANNEL_IO_NAME.contains(channelId)) {
+                return getStringState(state.ioName[index]);
+            }
+            if (IAnelConstants.CHANNEL_IO_STATE.contains(channelId)) {
+                return getSwitchState(state.ioState[index]);
+            }
+            if (IAnelConstants.CHANNEL_IO_MODE.contains(channelId)) {
+                return getSwitchState(state.ioState[index]);
+            }
+        } else {
+            if (IAnelConstants.CHANNEL_NAME.equals(channelId)) {
+                return getStringState(state.name);
+            }
+            if (IAnelConstants.CHANNEL_TEMPERATURE.equals(channelId)) {
+                return getDecimalState(state.temperature);
+            }
+
+            if (IAnelConstants.CHANNEL_SENSOR_TEMPERATURE.equals(channelId)) {
+                return getDecimalState(state.sensorTemperature);
```

Comment:
If you use Units of Measure, you need to use `QuantityType` instead of `DecimalType` to update the channel.